config/puma.rb

```diff
@@ -36,3 +37,7 @@
 
 # Allow puma to be restarted by `rails restart` command.
 plugin :tmp_restart
+
+if Settings.scc.try(:metrics) && (Settings.scc.metrics.try(:enabled) && (!!Settings.scc.metrics.enabled  == false))
+  configure_prometheus!(self)
+end
```

Comment:
Puma doesn't know about the `Setttings` object yet, this belongs to the application. Although it may work, this is because of the `Settings` class having default settings to operate.

If you further down in the app code change the default opts then you end up in inconsistent behavior, puma exclusively know about the default behavior but the app sees the altered behavior, although both use the same interface.

On puma config you should not interface with application internals because the application is yet to be loaded

Reply:
This is not true. `lib/rmt/config.rb` sets up `Settings` object.

Reply:
```
✗ bex puma 
Traceback (most recent call last):
        15: from /home/scc/.rvm/gems/ruby-2.5.9/bin/ruby_executable_hooks:22:in `<main>'
        14: from /home/scc/.rvm/gems/ruby-2.5.9/bin/ruby_executable_hooks:22:in `eval'
        13: from /home/scc/.rvm/gems/ruby-2.5.9/bin/puma:23:in `<main>'
        12: from /home/scc/.rvm/gems/ruby-2.5.9/bin/puma:23:in `load'
        11: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/bin/puma:8:in `<top (required)>'
        10: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/bin/puma:8:in `new'
         9: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/lib/puma/cli.rb:72:in `initialize'
         8: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/lib/puma/cli.rb:72:in `new'
         7: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/lib/puma/launcher.rb:60:in `initialize'
         6: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/lib/puma/configuration.rb:219:in `load'
         5: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/lib/puma/configuration.rb:219:in `each'
         4: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/lib/puma/configuration.rb:219:in `block in load'
         3: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/lib/puma/dsl.rb:87:in `_load_from'
         2: from /home/scc/.rvm/gems/ruby-2.5.9/gems/puma-5.6.8/lib/puma/dsl.rb:87:in `instance_eval'
         1: from config/puma.rb:1:in `_load_from'
config/puma.rb:1:in `require': cannot load such file -- rmt/config (LoadError)
```

As I said, puma doesn't know about this object yet because the application has not loaded.

When you run the webserver using `bin/rails server` the config files for puma are never loaded. The boot sequence first loads the `rails` framework, then it loads `rack` and then goes over `rackup` to start the server (based on puma or Webrick, nowadays is puma) and then the puma config may be loaded.

Reply:
I understand what you mean but rmt server is meant to be run with `rails s`, look at the `.service` file, or even the entrypoint script for the docker image. I think that was to make the server options configurable through `rmt.{conf,yml}`.

Reply:
Ach so... welp, that overrides completely my position then.